### PR TITLE
Support for custom HTTP Headers

### PR DESCRIPTION
**Pull request by 2moe**
# Pull Request

## Types of changes

- New feature (Support for custom HTTP Headers)
- Chore (Upgrade and simplify deps)

## Description

close https://github.com/rgreinho/trauma/issues/52

### About feature:

Added `.headers()` and `.header()` for `DownloaderBuilder`.
We can either call the `.header()` multiple times to add multiple headers, or we can add a bunch of `headers` with `.headers()`.

```rust
   use reqwest::{
            header::{self, HeaderValue},
        };

    const FIREFOX_UA: &str =
        "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/109.0";

    let ua = HeaderValue::from_str(FIREFOX_UA).expect("Invalid UA");

    let client = DownloaderBuilder::new()
        .directory(PathBuf::from("assets"))
        .header(header::USER_AGENT, ua)
        .build();
```

### About dependencies:

`http::header` -> `reqwest::header`
`url::Url` -> `reqwest::Url`

Reducing external dependencies makes maintenance easier, as well as reducing the probability of dependency conflicts.

## Checklist

- [x] I have updated the documentation accordingly
- [] I have updated the Changelog (if applicable)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "trauma"
-version = "2.1.1"
+version = "2.2.0"
```
Comment (rgreinho) [inline Cargo.toml:3]: Don't update the version number. We change it when we cut a new release.

**Review by rgreinho — CHANGES_REQUESTED** [review]
Looking good! Thank you for your contribution 👍 

Only one minor thing I would like you to change regarding the version number.

Also, for some reason the formatting of the PR seems off and causes `cargo fmt` to complain. Are you using Rust nightly?

File: Cargo.toml (hunk 1 above)
Comment (2moe) [reply Cargo.toml]: I am using rust stable.
The format is mainly related to rustfmt.
Can you provide the relevant configuration?

```json
    "rust-analyzer.rustfmt.extraArgs": [
        "--config",
        "max_width=85,single_line_if_else_max_width=80,chain_width=32,format_code_in_doc_comments=false,imports_granularity=Crate"
    ],
    "rust-analyzer.checkOnSave.command": "clippy",
```

**Review by rgreinho — APPROVED** [review]
👍

**Pull request merged**